### PR TITLE
Set dataloader.batch_size = None when batch_sampler is given

### PR DESCRIPTION
**Pull request by ssnl**
Set dataloader's `batch_size` and `drop_last` to `None` when `batch_sampler is given. Add a warning in doc for this case. 

Address https://github.com/pytorch/pytorch/issues/5884.

### CODE REVIEW

File: torch/utils/data/dataloader.py [hunk 1]
```diff
@@ -376,6 +376,12 @@ class DataLoader(object):
             worker subprocess with the worker id (an int in ``[0, num_workers - 1]``) as
             input, after seeding and before data loading. (default: None)
 
+    .. warning:: If :attr:`batch_sampler` is given, setting :attr:`batch_size`,
+                 :attr:`shuffle`, :attr:`sampler`, or :attr:`drop_last` is
+                 invalid because such information is provided by
+                 :attr:`batch_sampler`. In this case, the data loader will have
+                 ``dataloader.batch_size == dataloader.drop_last == dataloader.sampler == None``.
```
Comment (apaszke) [inline torch/utils/data/dataloader.py]: Can we implement a check in `__setattr__`?

**Review by ezyang — APPROVED** [review]

File: torch/utils/data/dataloader.py (hunk 1 above)
Comment (SsnL) [reply torch/utils/data/dataloader.py]: You are right.

**Review by apaszke — COMMENTED** [review]
LGTM, but let's remove the warning from the docs

**Review by apaszke — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**